### ISSUE TITLE
`azurerm_eventhub_namespace` issue with `public_network_access`  filed

### ISSUE THREAD

**Issue by sreekanth3107** [hashicorp/terraform-provider-azurerm#18485]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.2

### AzureRM Provider Version

3.23

### Affected Resource(s)/Data Source(s)

azurerm_eventhub_namespace

### Terraform Configuration Files

```hcl
resource "azurerm_eventhub_namespace" "example" {
  name                = "example-namespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1
  

  network_rulesets {
    default_action        = "Deny"
    virtual_network_rule  = []
    ip_rule               = []
    public_network_access = true
  }

  tags = {
    environment = "Production"
  }
}
```

### Debug Output/Panic Output

```shell
resource "azurerm_eventhub_namespace" "example":
public_network_access = true
An argument named "public_network_access" is not expected here
```

### Expected Behaviour

An argument named "public_network_access" is not expected here

### Important Factoids

 I tried removing the `public_network_access` filed from resource, that time i am seeing an issue like 

```
Namespace Name: "primary-eventhub-xxx"): networkrulesets.NetworkRuleSetsClient#NamespacesCreateOrUpdateNetworkRuleSet: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="This update of networkRuleSets, sets zero IPRule and VirtualNetworkRules, and the DefaultAction is set to Deny. This would render the namespace resources inaccessible for data-plane requests. If the intention was to restrict data-plane requests to private links only, please set the PublicNetworkAccess property to true. Else either set the DefaultAction property to Allow OR add some allowed IPRules or VirtualNetworkRules in the update request. CorrelationId: 8d1c4468-a628-e2fd-5c2d-e22cbf6f803d"
```

### References

The `public_network_access` is supported by latest version of provider. [Terraform doc](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventhub_namespace#public_network_access)

**Comment by xiaxyi** [comment]
Thanks @sreekanth3107 for raising this issue, can you try `public_network_access_enabled` instead of `public_network_access `?

I will modify the doc.

**Comment by xiaxyi** [comment]
@sreekanth3107 may I know the detailed issue?

**Comment by sreekanth3107** [comment]
@Gerrit-K  I don’t think there is any typo error. Please find the below configuration and versions :

 azurerm version     = "3.26.0" 

Resource Example: 
 
 ```
resource "azurerm_resource_group" "sree-example" {
  name     = "example-resources"
  location = "Central US"
}

resource "azurerm_eventhub_namespace" "example" {
  name                = "sree-example-namespace"
  location            = azurerm_resource_group.sree-example.location
  resource_group_name = azurerm_resource_group.sree-example.name
  sku                 = "Standard"
  capacity            = 2

  network_rulesets {
    default_action                = "Deny"
    virtual_network_rule          = []
    ip_rule                       = []
    public_network_access_enabled = null
  }
}
```

Terraform plan :

```
  # azurerm_eventhub_namespace.example will be created
  + resource "azurerm_eventhub_namespace" "example" {
      + auto_inflate_enabled                      = false
      + capacity                                  = 2
      + default_primary_connection_string         = (sensitive value)
      + default_primary_connection_string_alias   = (sensitive value)
      + default_primary_key                       = (sensitive value)
      + default_secondary_connection_string       = (sensitive value)
      + default_secondary_connection_string_alias = (sensitive value)
      + default_secondary_key                     = (sensitive value)
      + id                                        = (known after apply)
      + local_authentication_enabled              = true
      + location                                  = "centralus"
      + maximum_throughput_units                  = (known after apply)
      + minimum_tls_version                       = (known after apply)
      + name                                      = "sree-example-namespace"
      + network_rulesets                          = [
          + {
              + default_action                 = "Deny"
              + ip_rule                        = []
              + public_network_access_enabled  = true
              + trusted_service_access_enabled = null
              + virtual_network_rule           = null
            },
        ]
      + public_network_access_enabled             = true
      + resource_group_name                       = "example-resources"
      + sku                                       = "Standard"
      + zone_redundant                            = false
    }

  # azurerm_resource_group.sree-example will be created
  + resource "azurerm_resource_group" "sree-example" {
      + id       = (known after apply)
      + location = "centralus"
      + name     = "example-resources"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
```

Terraform apply 

```
azurerm_resource_group.sree-example: Creating...
azurerm_resource_group.sree-example: Creation complete after 1s [id=/subscriptions/xxxx/resourceGroups/example-resources]
azurerm_eventhub_namespace.example: Creating...
azurerm_eventhub_namespace.example: Still creating... [10s elapsed]
azurerm_eventhub_namespace.example: Still creating... [20s elapsed]
azurerm_eventhub_namespace.example: Still creating... [30s elapsed]
azurerm_eventhub_namespace.example: Still creating... [40s elapsed]
azurerm_eventhub_namespace.example: Still creating... [50s elapsed]
azurerm_eventhub_namespace.example: Still creating... [1m0s elapsed]
╷
│ Error: setting network ruleset properties for Namespace (Subscription: "xxxxxx"
│ Resource Group Name: "example-resources"
│ Namespace Name: "sree-example-namespace"): networkrulesets.NetworkRuleSetsClient#NamespacesCreateOrUpdateNetworkRuleSet: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="This update of networkRuleSets, sets zero IPRule and VirtualNetworkRules, and the DefaultAction is set to Deny. This would render the namespace resources inaccessible for data-plane requests. If the intention was to restrict data-plane requests to private links only, please set the PublicNetworkAccess property to Disabled. Else either set the DefaultAction property to Allow OR add some allowed IPRules or VirtualNetworkRules in the update request. CorrelationId: 417fb527-be6f-5ee5-7991-0abf28c98fcf"
│ 
│   with azurerm_eventhub_namespace.example,
│   on main.tf line 6, in resource "azurerm_eventhub_namespace" "example":
│    6: resource "azurerm_eventhub_namespace" "example" {
│ 
╵

```

**Comment by sreekanth3107** [comment]
@Gerrit-K I tired same thing with  `azurerm `version 3.26. Please find the confirmation below.

Terraform plan:

```
+ resource "azurerm_eventhub_namespace" "example" {
      + auto_inflate_enabled                      = false
      + capacity                                  = 2
      + default_primary_connection_string         = (sensitive value)
      + default_primary_connection_string_alias   = (sensitive value)
      + default_primary_key                       = (sensitive value)
      + default_secondary_connection_string       = (sensitive value)
      + default_secondary_connection_string_alias = (sensitive value)
      + default_secondary_key                     = (sensitive value)
      + id                                        = (known after apply)
      + local_authentication_enabled              = true
      + location                                  = "centralus"
      + maximum_throughput_units                  = (known after apply)
      + minimum_tls_version                       = (known after apply)
      + name                                      = "sree-example-namespace"
      + network_rulesets                          = [
          + {
              + default_action                 = "Allow"
              + ip_rule                        = []
              + public_network_access_enabled  = true
              + trusted_service_access_enabled = null
              + virtual_network_rule           = null
            },
        ]
      + public_network_access_enabled             = true
      + resource_group_name                       = "example-resources"
      + sku                                       = "Standard"
      + zone_redundant                            = false
    }
```
 
**Issues with Terraform Apply :**


```
│ Error: setting network ruleset properties for Namespace (Subscription: "xxxxxxxx"
│ Resource Group Name: "example-resources"
│ Namespace Name: "sree-example-namespace"): networkrulesets.NetworkRuleSetsClient#NamespacesCreateOrUpdateNetworkRuleSet: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="This update of networkRuleSets, sets zero IPRule and VirtualNetworkRules, and the DefaultAction is set to Deny. This would render the namespace resources inaccessible for data-plane requests. If the intention was to restrict data-plane requests to private links only, please set the PublicNetworkAccess property to Disabled. Else either set the DefaultAction property to Allow OR add some allowed IPRules or VirtualNetworkRules in the update request. CorrelationId: 7f8630e1-9692-7a5b-34b2-e64a1aa0258f"
│ 
│   with azurerm_eventhub_namespace.example,
│   on main.tf line 6, in resource "azurerm_eventhub_namespace" "example":
│    6: resource "azurerm_eventhub_namespace" "example" {
```

Working options :
- Option 1 : updating the `default_action = "Allow"` its works fine but this doesn’t help we need to make `Deny` all time.
- Option 2 : updating the `ip_rule = xx.xx.xx.xx/x` its works fine but we need to allow all IPS 

But we don’t to update  this fields not sure what changes in recent release version but if i use the  `azurerm `version 3.12.0 everything works fine?

**Comment by Gerrit-K** [comment]
@sreekanth3107 now that's a completely different error message as stated initially. And it's quite detailed as well. In addition to your options there's also option 3: set `public_network_access_enabled` to false.

However, since I don't know your exact business / product requirements, I can't tell which option is best for you :/

What I _can_ tell from experience, thoufh, is that API changes like these unfortunately happen, which then often leads to sudden pipeline or workflow failures. If you really want to know what's going on, I'd suggest to try to find the exact provider version update when this went wrong and then dig through the associated changes since the previous version.

**Comment by sreekanth3107** [comment]
@Gerrit-K When i open my issue initially i had issue with terraform plan like `public_network_access` and after updating filed names its work's fine. Thanks for that quick help.

I tried `public_network_access_enabled` = false as well that will block the all public IP'S which is not acceptable. I tried validating previous version but didn’t much help but i confidently says that this is happening after  version `3.17.0`
